client/trino-client/src/main/java/io/trino/client/auth/kerberos/GSSContextProvider.java

```diff
@@ -13,15 +13,28 @@
  */
 package io.trino.client.auth.kerberos;
 
+import io.airlift.units.Duration;
+import org.ietf.jgss.GSSContext;
 import org.ietf.jgss.GSSException;
+import org.ietf.jgss.Oid;
 
-import javax.security.auth.Subject;
 import javax.security.auth.login.LoginException;
 
-public interface SubjectProvider
+public interface GSSContextProvider
```

Comment:
Can we extract the `GSSContextProvider` into a seprate commit - and handle the delegation support as a follow-up ?

Reply:
Extracted to separate commit